float32 finger1
float32 finger2
float32 finger3
float32 isopen
